### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by mcardosos**
To be merge when https://github.com/Azure/azure-sdk-for-go/pull/784 is merged

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -36,16 +36,22 @@ imports:
   version: 5bf94b69c6b68ee1b541973bb8e1144db23a194b
 - name: github.com/shopspring/decimal
   version: 3c692774ac4c47c7a296ec96e553719dba1a68fc
+- name: github.com/xtophs/azure-sdk-for-go
```
Comment (jhendrixMSFT) [inline glide.lock]: Why is this here?

File: arm/monitor/servicediagnosticsettings.go [hunk 2]
```diff
@@ -1,242 +0,0 @@
-package monitor
```
Comment (jhendrixMSFT) [inline arm/monitor/servicediagnosticsettings.go]: Is it ok that this file is deleted?

File: arm/redis/firewallrule.go [hunk 3]
```diff
@@ -1,254 +0,0 @@
-package redis
```
Comment (jhendrixMSFT) [inline arm/redis/firewallrule.go]: Deleted.

File: arm/redis/firewallrules.go [hunk 4]
```diff
@@ -1,132 +0,0 @@
-package redis
```
Comment (jhendrixMSFT) [inline arm/redis/firewallrules.go]: Deleted.c

File: arm/redis/firewallrule.go (hunk 3 above)
Comment (mcardosos) [reply arm/redis/firewallrule.go]: Checked the redis swagger, firewall stuff is included in a [previous](https://github.com/Azure/azure-rest-api-specs/blob/current/specification/redis/resource-manager/Microsoft.Cache/2016-04-01/redis.json) version, but not in [latest](https://github.com/Azure/azure-rest-api-specs/tree/current/specification/redis/resource-manager/Microsoft.Cache/2017-02-01) version

File: glide.lock (hunk 1 above)
Comment (mcardosos) [reply glide.lock]: This should not be here, created PR
https://github.com/Azure/azure-sdk-for-go/pull/787

File: arm/redis/firewallrule.go (hunk 3 above)
Comment (jhendrixMSFT) [reply arm/redis/firewallrule.go]: I think we should move back to the previous version as this would be a pretty big breaking change and I see no mention that the firewall stuff has been deprecated (we should follow up with the redis cache folks to understand what's going on here).

**Comment by jhendrixMSFT** [comment]
Do we need to upgrade go-autorest to v8.4 to pick up the MSI work?

**Comment by mcardosos** [comment]
@jhendrixMSFT 
Included go-autorest update to 8.4 here https://github.com/Azure/azure-sdk-for-go/pull/787

File: arm/monitor/servicediagnosticsettings.go (hunk 2 above)
Comment (jhendrixMSFT) [reply arm/monitor/servicediagnosticsettings.go]: I think this file was renamed yes?

Comment (mcardosos) [reply arm/monitor/servicediagnosticsettings.go]: Still looking

File: arm/redis/firewallrule.go (hunk 3 above)
Comment (mcardosos) [reply arm/redis/firewallrule.go]: New PR generated with old version https://github.com/Azure/azure-sdk-for-go/pull/788

File: arm/monitor/servicediagnosticsettings.go (hunk 2 above)
Comment (mcardosos) [reply arm/monitor/servicediagnosticsettings.go]: Looks like operations moved to different operation group

File: arm/analysisservices/servers.go [hunk 5]
```diff
@@ -109,7 +106,7 @@ func (client ServersClient) CreatePreparer(resourceGroupName string, serverName
 		"subscriptionId":    autorest.Encode("path", client.SubscriptionID),
 	}
 
-	const APIVersion = "2016-05-16"
+	const APIVersion = "2017-08-01-beta"
```
[the remark this answers is not in the document]
Comment (mcardosos) [reply arm/analysisservices/servers.go]: Comes from the swagger repo readme file. We already had generated the SDK with this version

File: arm/billing/periods.go [hunk 6]
```diff
@@ -14,9 +14,8 @@ package billing
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-// Code generated by Microsoft (R) AutoRest Code Generator 1.0.1.0
-// Changes may cause incorrect behavior and will be lost if the code is
-// regenerated.
+// Code generated by Microsoft (R) AutoRest Code Generator 2.2.22.0
```
[the remark this answers is not in the document]
Comment (mcardosos) [reply arm/billing/periods.go]: Fixed in PR https://github.com/Azure/azure-sdk-for-go/pull/788
